ecosystem/auto-relayer/auto-relayer.md

```diff
@@ -0,0 +1,210 @@
+
+### Wallet Management
+The following approach will be used to keep wallets funded on the relayer:
+1. A Drippie contract is deployed and funded on each network in the interop set
+2. Drips on the Drippie contract are configured for each wallet, so that when the balance of the wallet is below x amount, Drippie will send funds to it
+3. The Gas Refund Processor has a cron job configured on it to regularly call the Drippie contract and trigger drips
+
+In order to make sure that this is running efficiently the following will be monitored:
+- Balance on each Drippie contract
+- Balance on each wallet
+- Drip transaction failures
+
+This approach provides easier management of funds because only one contract on each chain needs to remain funded instead of having to individually fund each wallet used by the relayer.
+
+### Logging and Alerting
+The following metrics will be used in order to track the overall health of the relayer and to configure alerting:
+- count of successful relays
+- count of successful refund claims
+- count of sent messages
+- count of failed relays
+- count of failed refund claims
+- balance of each wallet
+- storage available on indexer db
+- memory available on tx relayer processor redis
+- memory available on gas refund processor redis
+- time between message sent and message relayed
+- balance of Drippie contracts
+
+### Gas Price
+**_Open question: do we need to allow a max gas price to be specified per message?_**
+
+**_Open question: are we okay with the relayer paying the priority fee and not being refunded for it?_**
+
+The gas fee charged by the relayer is based on `block.basefee` and not the `tx.gasprice` (see [#266](https://github.com/ethereum-optimism/design-docs/pull/266/files)), which means that the priority fee set by the relayer will not be charged to the user. In later iterations of this service, we will consider exploring a priority fee mechanism, where users can leverage a priority fee in order to increase the speed at which their messages are relayed.
+
+### Testing
+Since the relayer is a production critical service, it is important that we have a high degree of confidence that the changes  pushed to it do not break the service. As a part of this effort we will need to run integration tests that spin up the relayer against [kurtosis interop devnets](https://github.com/ethereum-optimism/optimism/tree/develop/kurtosis-devnet).
+
+## Failure Mode Analysis
```

Comment:
Some of this is covered above, like how stuck transactions are handled. But just wanted to share my list of what I think should be covered in FMA in case its scheduled while I'm out:

- redis outage
- postgres lag
- stuck transactions
- funded wallet is empty
- chain reorg
- chain halt
- making sure keys are stored properly and we're rotating them